test/test_other.py

```diff
@@ -1306,6 +1306,42 @@ def test_export_all(self):
     self.emcc('lib.c', ['-Oz', '-sEXPORT_ALL', '-sLINKABLE', '--pre-js', 'main.js'], output_filename='a.out.js')
     self.assertContained('libf1\nlibf2\n', self.run_js('a.out.js'))
 
+  def test_export_all_modularize(self):
+    """
+    This test ensures that modularize and export all work simulatenously.
+
+    In addition, it ensures that EXPORT_ALL is honored while using MINIMAL_RUNTIME.
+    """
+
+    main = r'''
+      #include <stdio.h>
+      #include <emscripten.h>
+      EMSCRIPTEN_KEEPALIVE void libf1() { printf("libf1\n"); }
+      EMSCRIPTEN_KEEPALIVE void libf2() { printf("libf2\n"); }
+    '''
+    create_file('main.c', main)
+
+    self.emcc('main.c', ['-sMODULARIZE=1', '-sMINIMAL_RUNTIME=2', '-sEXPORT_ALL', '-sEXPORT_ES6'], output_filename='test.mjs')
+
+    # We must expose __dirname and require globally because emscripten
+    # uses those under the hood.
```

Comment:
Why is this an issue in this test but not in general?

Reply:
Basically this test runs in nodejs and import a module, however using `MODULARIZE` and `MINIMAL_RUNTIME`, the UMD export is disabled. I exported using `ES6` and it looks like this doesn't run directly in nodejs due to the internal code using require (not available in an ES6 module in node by default).

I did a quick search and the other tests are either running in the web, or are just checking the content of the file written.

If there is a better way to run the test, maybe in the browser, and get the output, I don't mind doing so.

Reply:
> I did a quick search and the other tests are either running in the web, or are just checking the content of the file written.

"the other tests" - which subset of tests do you mean? (99% of our tests run in node, so you must need some specific feature here?)

Reply:
The tests already written don't do `MODULARIZE` + `MINIMAL_RUNTIME` + `EXPORT_ES6` in node AFAIK, if I am wrong can you please point me to one?
The tests that mix `MODULARIZE` and `MINIMAL_RUNTIME` do not import nor require the exported module (example of test like that: `browser.test_minimal_runtime_modularize`)

Reply:
Yes, that looks like current situation.  We don't test `MODULARIZE` + `MINIMAL_RUNTIME` except in browser tests.  I was looking at this just yesterday.  Not sure how hard it would be to do some testing under node too..  not completely trivial but probably worth looking into

Reply:
Good ideas. Yes, it would be nice to test modularize + minimal runtime outside of the browser, and maybe we can extract some helper code for that from this test later.